src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php

```diff
@@ -37,17 +38,35 @@ public function create(ContainerBuilder $container, string $id, array|string $co
 
         // @see Symfony\Bundle\SecurityBundle\DependencyInjection\Security\Factory\SignatureAlgorithmFactory
         // for supported algorithms
-        if (\in_array($config['algorithm'], ['ES256', 'ES384', 'ES512'], true)) {
+        if (\in_array($config['algorithm'], ['ES256', 'ES384', 'ES512', 'RS256'], true)) {
             $tokenHandlerDefinition->replaceArgument(0, new Reference('security.access_token_handler.oidc.signature.'.$config['algorithm']));
         } else {
             $tokenHandlerDefinition->replaceArgument(0, (new ChildDefinition('security.access_token_handler.oidc.signature'))
                 ->replaceArgument(0, $config['algorithm'])
             );
         }
 
-        $tokenHandlerDefinition->replaceArgument(1, (new ChildDefinition('security.access_token_handler.oidc.jwk'))
-            ->replaceArgument(0, $config['key'])
-        );
+        if (!isset($config['jwks_url']) && !isset($config['key'])) {
+            throw new LogicException('You should defined key or jwks_url parameter in configuration.');
+        }
+
+        if (isset($config['jwks_url'])) {
+            if (!class_exists(HttpClient::class)) {
+                throw new LogicException(sprintf('You cannot use "%s" as the HttpClient component is not installed. Try running "composer require symfony/http-client".', __CLASS__));
+            }
+            $httpClient = HttpClient::create();
```

Comment:
This cannot be the way because creating the client inline breaks IoC and also means we're going to hit the oidc server for every single requests.
I suggest toremove the "url" part from this PR so that we can consider how this request should be done separately,and keep this PR on adding support for JWKSet only.

Reply:
I removed the code for the support of JwkSet in SecurityBundle.  
I think we can do this in another PR ?